### PR TITLE
Fix vispr

### PR DESCRIPTION
**Pull request by corburn**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/vispr/meta.yaml [hunk 1]
```diff
@@ -9,25 +9,25 @@ source:
   sha256: dc2cf6ac9c8930b0f1f1b3a2b7f57bfae180b2ac3674060123fd2d7fe10ccd82
 
 build:
-  skip: True # [py27]
+  noarch: python
```
Comment (bgruening) [inline recipes/vispr/meta.yaml]: noarch python should only be used by pure python packages, numpy is not noarch and not pure python my understanding is that this is not working. Maybe @mbargull knows more?

Comment (mbargull) [reply recipes/vispr/meta.yaml]: A `noarch` package can have non-`noarch` dependencies, no problem. A good example is the dependency on `python` itself, which is not `noarch` :wink:.
The package itself seems to be pure Python, so this looks good.

File: recipes/vispr/meta.yaml [hunk 2]
```diff
@@ -9,25 +9,19 @@ source:
...
-    - pyyaml
-    - scipy
-    - scikit-learn
-    - appdirs
+    - pip
   run:
-    - python
-    - numpy
+    - python >=3
+    - numpy >=1.10
```
Comment (mbargull) [inline recipes/vispr/meta.yaml]: @corburn, is there any reason you added those constraints?
Acc. to https://bitbucket.org/liulab/vispr/src/03dd84177d431f82943bec0b25e1540d4e674e6e/setup.py this should be Python 2 compatible (although it was `skip`ped before in the recipe) and no minimum `numpy` version is specified.

**Comment by mbargull** [comment]
Oh and BTW, @corburn, I don't think I mentioned it before, but: Thanks for all the help! Really appreciated!

File: recipes/vispr/meta.yaml (hunk 1 above)
Comment (mbargull) [reply recipes/vispr/meta.yaml]: For Python projects I usually just take a look at the `setup.py` and `setup.cfg` (+ `requirement.txt` if that one is used in `setup.py` and the license file of course).
If `setup.py` has no platform-specific code (including in requirement specifiers) and doesn't build any extensions, the package should usually be pure Python and thus be likely eligible for `noarch: python`.
(Of course there can be cases where `noarch: python` isn't really true, e.g., when \[only for a subset of all covered systems\] some system libraries are dynamically loaded or system-specific file system accesses happen -- but that would require knowing/reading the whole source code ;P)

Comment (mbargull) [reply recipes/vispr/meta.yaml]: And if the package's `setup.py` is nice (like it's the case for `vispr`) and only imports system modules but not itself or thirdparty packages, we can shrink the `build` and `requirements/host` section to
```yaml
build:
  script: python -m pip install --no-deps --ignore-installed .

requirements:
  host:
   - python
   - pip

File: recipes/vispr/meta.yaml (hunk 2 above)
Comment (corburn) [reply recipes/vispr/meta.yaml]: @mbargull 

The python constraint replaced the `skip` in the recipe.
The numpy constraint resolved this error:

```
Returning non-final recipe for vispr-0.4.14-py36_2; one or more dependencies was unsatisfiable:^M                                                                                                                  
Build: None^M                                                                                                                                                                                                      
Host: The following specifications were found to be in conflict:^M                                                                                                                                                 
  - numpy=1.9^M                                                                                                                                                                                                    
  - pandas[version='>=0.17.1,<0.20'] -> numpy=1.10
```

**Pull request merged**